### PR TITLE
[Security] Do not use First Class Callable Syntax for listeners

### PR DESCRIPTION
**Pull request by javer**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

This PR fixes usage of First Class Callable Syntax for registering listeners which are intended to be unregistered later.

I'm targeting 6.1 branch because the issue was introduced only in this branch via https://github.com/symfony/symfony/commit/b0217c600f078d58d6c3bd9895d6a015d1f24c14

The reason of this change is clear from the following example:
```php
<?php

class Foo {
    public function bar() {}
}

$foo = new Foo();
var_dump($foo->bar(...) === $foo->bar(...)); // false
var_dump([$foo, 'bar'] === [$foo, 'bar']);   // true
```

In other words, it's not safe to use First Class Callable Syntax in the cases when we need to refer to the same function to remove it from somewhere, because each `method(...)` returns a new function.

### CODE REVIEW

**Comment by javer** [comment]
Also it seems that CI is broken:
```
Can't contact LDAP server (-1)
Uncaught UnexpectedValueException: Intersection types not yet supported
Symfony\Bundle\WebProfilerBundle\Tests\Resources\IconTest::testIconFileContents with data set #22516
```
and so on.

**Review by stof — APPROVED** [review]

**Review by nicolas-grekas — COMMENTED** [review]
Good catch. I don't think this is the patch we want to merge. Instead, remove listener should be able to work with first-class callables. I'm going to submit a PR doing so, so that we can discuss about it.

**Comment by nicolas-grekas** [comment]
See #46262

**Comment by chalasr** [comment]
Closing in favor of #46262 which reuses the test cases you contributed. Thank you.
